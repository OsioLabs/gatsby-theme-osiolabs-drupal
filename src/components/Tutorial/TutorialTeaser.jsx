import React from 'react';
import PropTypes from 'prop-types';
import Prism from 'prismjs';
import { Container, Header } from 'semantic-ui-react';

// Styling for code syntax highlighting.
require('prismjs/themes/prism.css');

const TutorialTeaser = ({ title, summary }) => (
  <Container>
    <Header as="h1">{title}</Header>
    <div dangerouslySetInnerHTML={{ __html: summary }} />
  </Container>
);

TutorialTeaser.componentDidMount = () => {
  // Whenever the component updates trigger Prism so that any new content will
  // get syntax highlighting.
  Prism.highlightAll();
};

TutorialTeaser.propTypes = {
  title: PropTypes.string.isRequired,
  summary: PropTypes.string.isRequired,
  changed: PropTypes.instanceOf(Date).isRequired,
};

export default TutorialTeaser;
