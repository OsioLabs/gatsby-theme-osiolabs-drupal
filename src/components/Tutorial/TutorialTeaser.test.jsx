import React from 'react';
import { render } from '@testing-library/react';

import TutorialTeaser from './TutorialTeaser';

describe('Component: <TutorialTeaser />', () => {
  it('renders correctly', () => {
    // Basic snapshot test.
    const { asFragment } = render(
      <TutorialTeaser
        title="Tacos Rock"
        summary="<p>Tacos <strong>go here</strong> ...</p>"
        changed={new Date()}
      />
    );
    const firstRender = asFragment();
    expect(firstRender).toMatchSnapshot();
  });
});
