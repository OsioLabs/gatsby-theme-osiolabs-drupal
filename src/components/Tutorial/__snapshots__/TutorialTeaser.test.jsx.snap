// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: <TutorialTeaser /> renders correctly 1`] = `
<DocumentFragment>
  <div
    class="ui container"
  >
    <h1
      class="ui header"
    >
      Tacos Rock
    </h1>
    <div>
      <p>
        Tacos 
        <strong>
          go here
        </strong>
         ...
      </p>
    </div>
  </div>
</DocumentFragment>
`;
